draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -218,12 +218,40 @@ Each member zone MAY have one or more additional properties, described in this c
 These properties are completely optional and the catalog zone consumer SHOULD ignore those it does not understand.
 Properties are represented by RRsets below the corresponding member node.
 
+## The Change of ownership (Coo) Property
+
+The 'coo' property facilitates controlled migration of a member zone from one catalog to another.
+
+A Change Of Ownership is signaled by the 'coo' property in the catalog zone currently ``owning'' the zone.
+The name of the new catalog is in the value of a PTR record in the old catalog.
+For example if member "example.com." will migrate from catalog zone `$OLDCATZ` to catalog zone `$NEWCATZ`, this appears in the `$OLDCATZ` catalog zone as follows:
+
+```
+<unique-N>.zones.$OLDCATZ 0 IN PTR example.com.
+coo.<unique-N>.zones.$OLDCATZ 0 IN PTR zones.$NEWCATZ
+```
+
+The PTR RRset MUST consist of a single PTR record.
+A catalog zone consumer MUST not process and ignore PTR RRsets with more than a single record.
+
+When a catalog zone consumer of catalog zone `$OLDCATZ` receives an update which adds or changes a `coo` property for a member zone in `$OLDCATZ` signalling a new owner `$NEWCATZ`, it does *not* migrate the member zone immediately.
+This is because the catalog zone consumer does not have the `<unique-N>` identifier associated with the member zone in `$NEWCATZ` and because name servers do not index Resource Records by RDATA, it does not know wether or not the member zone is configured in `$NEWCATZ` at all.
+It has to wait for an update of `$NEWCATZ` adding or changing that member zone.
```

Comment:
This would not be the case if `<unique-N>` was predictable.

Reply:
Indeed, I'll change "does not have the `<unique-N>`" into "may not have the `<unique-N>`"

Reply:
I think the remainder of these two sentences are also not applicable when `<unique-N>` was predictable, because in that case, catalog entries would effectively be indexed by RDATA (namely, most plausibly, the hash of the PTR target), and the consumer would also not have to wait.

If you like, I can rewrite this to be a bit more general.

But the point I really wanted to make is that this coo business is really another reason for predictable labels, at least if you want it to run smoothly and without waiting for arbitrary times until the new catalog owner decides to do an update.

(There are more complications. For example, while "transfer-out" is blocked due to unknown new label: what if one of the properties (serial, group, ...) in the old catalog changes? Is the old catalog still "authoritative" for the member? etc.pp.)